### PR TITLE
Fix: Bug in MinMax which returned an incorrect simplication which is not true for all ranges of the variables

### LINKED ISSUE #21399: nested Max and Min give incorrect result
While working with different types of nested Max and Min expressions, I have encountered a problem in the following expression:
```
from sympy import sympify 
sympify( "Max( Min(a, b), Min(a, b, c))" )
```

output: `Min(a, b, c)`

This simplification is not correct for all possible values of a, b, and c.
I have not found any problems with other similar expressions, and I cannot find the specific reason why this one evaluates incorrectly.

### PR DESCRIPTION
**Pull request by shashankks0987**
#### References to other Issues or PRs

Fixes #21399 

#### Brief description of what is fixed or changed
earlier, Max( Min(a, b), Min(a, b, c)) returned Min(a,b,c)
now, Max( Min(a, b), Min(a, b, c)) returns Min(a,b)

- Add test for the above fix
- Add the above example in doc

#### Release Notes

* functions
   * Fixed a bug in MaxMin of elementary functions

### CODE REVIEW

File: sympy/functions/elementary/miscellaneous.py [hunk 1]
```diff
@@ -432,7 +433,8 @@ def _collapse_arguments(cls, args, **assumptions):
 
         >>> Min(a, Max(b, Min(c, d, Max(a, e))))
         Min(a, Max(b, Min(a, c, d)))
-
+        >>> Max( Min(a, b), Min(a, b, c))
+        Min(a,b)
```
Comment (oscarbenjamin) [inline sympy/functions/elementary/miscellaneous.py:437]: I don't think that an example needs to be added here.

File: sympy/functions/elementary/tests/test_miscellaneous.py [hunk 2]
```diff
@@ -410,6 +410,10 @@ def test_issue_12638():
     assert Min(a, b, Max(a, b, c)) == Min(a, b)
     assert Min(a, b, Max(a, c)) == Min(a, b)
 
+def test_issue_21399():
+    from sympy.abc import a, b, c
+    assert Max( Min(a, b), Min(a, b, c)) == Min(a,b)
```
Comment (oscarbenjamin) [inline sympy/functions/elementary/tests/test_miscellaneous.py:415]: The spacing here is off. There should be no space after `(` but there should be a space after `,` so:
```
assert Max(Min(a, b), Min(a, b, c)) == Min(a, b)
```

Comment (oscarbenjamin) [reply sympy/functions/elementary/tests/test_miscellaneous.py:415]: Also the line spacing between test functions here should be two blank lines between each function. Can you fix that for this function and also those below?

Comment (smichr) [inline sympy/functions/elementary/tests/test_miscellaneous.py:415]:
```suggestion
    assert Max(Min(a, b), Min(a, b, c)) == Min(a, b)
```

File: sympy/functions/elementary/miscellaneous.py (hunk 1 above)
Comment (smichr) [inline sympy/functions/elementary/miscellaneous.py:437]:
```suggestion
```

**Comment by smichr** [comment]
Thanks, looks good.

**Comment by shashankks0987** [comment]
https://github.com/sympy/sympy/pull/21547/checks?check_run_id=2710942124 was this test always failing?

**Pull request merged**